products/bluebell/src/bluebell.lalrpop

```diff
@@ -0,0 +1,1330 @@
+        "[" => lexer::Token::OpenBracket,
+        "]" => lexer::Token::CloseBracket,
+        "(" => lexer::Token::OpenParen,
+        ")" => lexer::Token::CloseParen,
+        "{" => lexer::Token::OpenBrace,
+        "}" => lexer::Token::CloseBrace,
+        "," => lexer::Token::Comma,
+        "=>" => lexer::Token::DoubleArrow,
+        "->" => lexer::Token::Arrow,
+        "=" => lexer::Token::Equals,
+        "&" => lexer::Token::Ampersand,
+        "<-" => lexer::Token::LeftArrow,
+        ":=" => lexer::Token::ColonEquals,
+        "@" => lexer::Token::At,
+        "-" => lexer::Token::Minus,
+        "_" => lexer::Token::Underscore,
+        "forall" => lexer::Token::Forall,
+        "builtin" => lexer::Token::Builtin,
+        "library" => lexer::Token::Library,
+        "import" => lexer::Token::Import,
+        "let" => lexer::Token::Let,
+        "in" => lexer::Token::In,
+        "match" => lexer::Token::Match,
+        "with" => lexer::Token::With,
+        "end" => lexer::Token::End,
+        "fun" => lexer::Token::Fun,
+        "tfun" => lexer::Token::Tfun,
+        "contract" => lexer::Token::Contract,
+        "transition" => lexer::Token::Transition,
+        "send" => lexer::Token::Send,
+        "event" => lexer::Token::Event,
+        "Event" => lexer::Token::EventType,
+        "field" => lexer::Token::Field,
+        "accept" => lexer::Token::Accept,
+        "exists" => lexer::Token::Exists,
+        "delete" => lexer::Token::Delete,
+        "throw" => lexer::Token::Throw,
+        "Map" => lexer::Token::Map,
+        "scilla_version" => lexer::Token::ScillaVersion,
+        "type" => lexer::Token::Type,
+        "of" => lexer::Token::Of,
+        "as" => lexer::Token::As,
+        "procedure" => lexer::Token::Procedure,
+        "Emp" => lexer::Token::Emp,
+        "ByStr" => lexer::Token::ByStr,
+
+        r"ByStr[0-9]+" => lexer::Token::ByStrWithSize(<&'input str>), 
+        
+
+        r"[+-]?[0-9]+" => lexer::Token::Number(<&'input str>), 
+        r"0(x|X)([a-fA-F0-9][a-fA-F0-9])*" => lexer::Token::HexNumber(<&'input str>), 
+        r#""(?:\\.|[^"])*""# => lexer::Token::StringLiteral(<&'input str>), 
+
+        r"[a-z][a-zA-Z0-9_]*" => lexer::Token::Identifier(<&'input str>), 
+        r"[_][a-zA-Z0-9_]*" => lexer::Token::SpecialIdentifier(<&'input str>), 
+        r"[A-Z][a-zA-Z0-9_]*" => lexer::Token::CustomIdentifier(<&'input str>), 
+        r"['][A-Z][a-zA-Z0-9_]*" => lexer::Token::TemplateIdentifier(<&'input str>), 
+    }
+}
+
+
+
+
+
+// The rule that defines a byte string type in Scilla.
+// A byte string type in Scilla is a built-in type that represents a sequence of bytes.
+// The `BYSTR` keyword is used to define a byte string type in Scilla.
+//
+// <typeByteStr> ::= "ByStr" <digit>+
+//
+// This rule allows for byte string types to be defined in a Scilla contract.
+//
+// @syntax typeByteStr
+// @return A byte string type in Scilla.
+pub ByteString : NodeByteStr = {
```

Comment:
To answer your question in more detail, yes it is duplicated and we probably should sort it out as a constant. However, note that I am still not completely understanding how this works since the regex pattern is not really used by the parser. Rather it appears that these are used as identifiers.